### PR TITLE
Fix window view parser

### PR DESCRIPTION
**Pull request by Vxider**
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Fix window view parser

Detailed description / Documentation draft:

Fix window view bugs for the following cases:

1. use `TUMBLE_START` with function `now()`, like 
```
CREATE WINDOW VIEW wv AS SELECT count(a), TUMBLE_START(wid) AS w_start, TUMBLE_END(TUMBLE(now(), INTERVAL '3' SECOND)) AS w_end FROM mt GROUP BY TUMBLE(now(), INTERVAL '3' SECOND) AS wid;
```

2. Reuse tumble function in one query, like
```
CREATE WINDOW VIEW wv AS SELECT count(a), TUMBLE_START(TUMBLE(timestamp, INTERVAL '3' SECOND)) AS w_start FROM mt GROUP BY TUMBLE(timestamp, INTERVAL '3' SECOND) AS wid;
```

3. `WATCH` for window view with function `now()`
4. `time_attr` check of window function when used in WindowView. The first argument of window function cannot be a constant value (`toDataTime`) when used in WindowView, like
```
CREATE WINDOW VIEW wv AS SELECT count(a) FROM mt GROUP BY TUMBLE(toDateTime('2018-01-01 00:00:00'), INTERVAL '3' SECOND) AS wid;
```
 or will cause an exception when creating `inner_table`, we have manually added a column when creating `inner_table` with function `now()`(the following lines), but I think we do not need to support `toDateTime` because constant `time_attr` is meaningless in WindowView
https://github.com/ClickHouse/ClickHouse/blob/c522c06755bfa280b14bab4ab9c3f0a0f3d227f5/src/Storages/WindowView/StorageWindowView.cpp#L571-L579

### CODE REVIEW

File: src/Storages/WindowView/StorageWindowView.h [hunk 1]
```diff
@@ -150,6 +150,7 @@ class StorageWindowView final : public shared_ptr_helper<StorageWindowView>, pub
 private:
     Poco::Logger * log;
 
+    ASTPtr select_query;
     ASTPtr mergeable_query;
     ASTPtr final_query;
```
Comment (kssenii) [inline src/Storages/WindowView/StorageWindowView.h:155]: Please add a comment in this file about the differencies between these three variables. Now it requires to reread all sequence of visitors and modifications to remember the difference.

**Review by kssenii — APPROVED** [review]

File: src/Storages/WindowView/StorageWindowView.cpp [hunk 2]
```diff
@@ -89,22 +89,27 @@ namespace
                 {
                     data.is_tumble = t->name == "TUMBLE";
                     data.is_hop = t->name == "HOP";
+                    auto temp_node = t->clone();
+                    temp_node->setAlias("");
```
[the remark this answers is not in the document]
Comment (Vxider) [reply src/Storages/WindowView/StorageWindowView.cpp:93]: We clear the alias to find whether there are different window functions in windowview.

In this case
```
CREATE WINDOW VIEW test_01048.wv AS
SELECT
    count(a) AS count,
    TUMBLE(timestamp, toIntervalSecond('1')) AS wid
FROM test_01048.mt
GROUP BY TUMBLE(timestamp, toIntervalSecond('1'))

```
the  `temp_node: TUMBLE(timestamp, toIntervalSecond('1'))` and `data.serialized_window_function: TUMBLE(timestamp, toIntervalSecond('1')) AS wid` will not equal if alias has not reset to empty.

**Comment by kssenii** [comment]
btw, unfortunately window view tests still randomly fail in different PRs https://clickhouse-test-reports.s3.yandex.net/32256/a98f740bf21440a5f521add95867902d11ccbc99/fast_test.html#fail1 (two tests at one `01064_window_view_event_hop_to_bounded` and `01052_window_view_proc_tumble_to_now`)
so I will have to disable window view tests right now from our CI (https://github.com/ClickHouse/ClickHouse/pull/32257). If you have an idea how to make them stable, would be great to make a PR with reverting tests disabling.

**Comment by Vxider** [comment]
The `SELECT sleep(1);` in test `01064_window_view_event_hop_to_bounded` might cause the random fail. If the system froze, the results might not be pushed into table `dst` in time. The `01064_window_view_event_hop_to_bounded` might missing the last line because of this.

We can reduce the probability of this random fail by increasing sleep time in tests, but it will also slow down the tests. 

Or can we have a mechanism to detect if new data has been inserted in one table in tests?

**Comment by Vxider** [comment]
The `SELECT sleep(1);` in test `01064_window_view_event_hop_to_bounded` might cause the random fail. If the system froze, the results might not be pushed into table `dst` in time. The `01064_window_view_event_hop_to_bounded` might missing the last line because of this.

We can reduce the probability of this random fail by increasing sleep time in tests, but it will also slow down the tests. 

Or can we have a mechanism to detect if new data has been inserted in one table in tests?

**Comment by kssenii** [comment]
>  but it will also slow down the tests.

This is Ok, we have many long tests and also tests a run in parallel, so an increase does not make much difference.

> Or can we have a mechanism to detect if new data has been inserted in one table in tests?

No. Also may be using `.sh` test with retries in loop could be more helpful than `sql` tests.

**Comment by kssenii** [comment]
btw you need to merge with master to delete `Tags: disabled` from window view tests. Now window view tests are marked as disabled in master because I merged PR with disabling them this morning.

**Comment by Vxider** [comment]
I have enabled the window view tests. If the tests are still unstable, we can disable them again.

**Pull request merged**